### PR TITLE
feat: Update "Swap to ZETA" tutorial to "Swap to Any Token"

### PR DESCRIPTION
**Pull request by lukema95**
issue: #180

### CODE REVIEW

**Comment by lukema95** [comment]
# Swap tBNB(BSC Testnet) -> sUSDC(Sepolia Testnet) and no withdraw
```bash
npx hardhat interact \
--contract 0x7fe0A7f3Fb57BD69b2210dd075a44246d9Fc2eDD \
--network bsc_testnet \
--amount 0.01 \
--target-token 0xcC683A782f4B30c138787CB5576a86AF66fdc31d	 \
--recipient 0x6093537Aa6C8C8bf4705bda40aC9193977208B39 \
--withdraw false

🚀 Successfully broadcasted a token transfer transaction on bsc_testnet network.
📝 Transaction hash: 0x356bf7aefb823783ffe92eea79957d0f682c2eecf1ba4b0edccf5b6c33e3c76d

npx hardhat cctx 0x356bf7aefb823783ffe92eea79957d0f682c2eecf1ba4b0edccf5b6c33e3c76d                                                                  
✓ CCTXs on ZetaChain found.

✓ 0x24845bd91ad63abadb3231a88472f4efa73f0a3120c30f54805d06289cef923b: 97 → 7001: OutboundMined (Remote omnichain contract call completed)
```

# Swap tBNB(BSC Testnet) -> sUSDC(Sepolia Testnet) and withdraw
```bash
npx hardhat interact \
--contract 0x7fe0A7f3Fb57BD69b2210dd075a44246d9Fc2eDD \
--network bsc_testnet \
--amount 0.1 \
--target-token 0xcC683A782f4B30c138787CB5576a86AF66fdc31d	 \
--recipient 0x6093537Aa6C8C8bf4705bda40aC9193977208B39

🚀 Successfully broadcasted a token transfer transaction on bsc_testnet network.
📝 Transaction hash: 0x29a294ebb021c5860bce4f071d64530f3387e161268c790717f4d03c13f19e86

npx hardhat cctx 0x29a294ebb021c5860bce4f071d64530f3387e161268c790717f4d03c13f19e86                                                                  
✓ CCTXs on ZetaChain found.

✓ 0x7177cd44d4487d9799bb5d976bfd13a5b614d7a86b87017c3ad6bb87e7109300: 97 → 7001: OutboundMined (Remote omnichain contract call completed)
✓ 0x33999e5e4f739fa2f230cbc5aecc0c7f61901696e9fa8c3dd1cd703466266ce6: 7001 → 11155111: PendingOutbound (ZRC20 withdrawal event setting to pending outbound directly) → OutboundMined (ZRC20 withdrawal event setting to pending outbound directly : Outbound succeeded, mined)
```

File: omnichain/swap/contracts/SwapToAnyToken.sol [hunk 1]
```diff
@@ -59,16 +70,25 @@ contract SwapToZeta is zContract, OnlySystem {
...
-            isTargetZeta ? amount : amount - inputForGas,
-            target,
+            params.withdraw ? amount - inputForGas : amount,
+            params.target,
             0
         );
 
-        if (isTargetZeta) {
-            IWETH9(wzeta).transfer(address(uint160(bytes20(to))), outputAmount);
+        if (!params.withdraw) {
```
Comment (fadeev) [inline omnichain/swap/contracts/SwapToAnyToken.sol:78]:
```suggestion
        if (params.withdraw) {
```

I think it's easier to read without negation.

**Comment by lukema95** [comment]
pls review @fadeev @andresaiello

File: omnichain/swap/contracts/SwapToAnyToken.sol [hunk 2]
```diff
@@ -59,16 +70,25 @@ contract SwapToZeta is zContract, OnlySystem {
...
+            IWETH9(params.target).transfer(address(uint160(bytes20(params.to))), outputAmount);
         }
     }
+
+    function bytesToBool(bytes calldata data, uint256 offset)
+        internal
+        pure
+        returns (bool)
+    {
+        require(offset < data.length, "Offset is out of bounds");
```
Comment (andresaiello) [inline omnichain/swap/contracts/SwapToAnyToken.sol:91]: move to custom error

**Comment by fadeev** [comment]
@lukema95 https://github.com/zeta-chain/toolkit/releases/tag/v10.0.0 has been released, which includes `bytesToBool`:

https://github.com/zeta-chain/toolkit/blob/d952cd7b67526c218e07030263d0d5413ab0e64c/contracts/BytesHelperLib.sol#L54-L63

If you can import it from the toolkit that'd be great!

File: omnichain/swap/contracts/SwapToAnyToken.sol (hunk 2 above)
Comment (fadeev) [reply omnichain/swap/contracts/SwapToAnyToken.sol:91]: This function will be removed and imported from the toolkit:

https://github.com/zeta-chain/toolkit/blob/d952cd7b67526c218e07030263d0d5413ab0e64c/contracts/BytesHelperLib.sol#L54-L63

**Comment by lukema95** [comment]
> @lukema95 https://github.com/zeta-chain/toolkit/releases/tag/v10.0.0 has been released, which includes `bytesToBool`:
> 
> https://github.com/zeta-chain/toolkit/blob/d952cd7b67526c218e07030263d0d5413ab0e64c/contracts/BytesHelperLib.sol#L54-L63
> 
> If you can import it from the toolkit that'd be great!

Looks like it hasn't been uploaded to the npm warehouse yet.

**Comment by fadeev** [comment]
@lukema95 I'm looking into it.

**Comment by fadeev** [comment]
@lukema95 v10.0.0 has been published on npm:

https://www.npmjs.com/package/@zetachain/toolkit/v/10.0.0

**Comment by lukema95** [comment]
> @lukema95 v10.0.0 has been published on npm:
> 
> https://www.npmjs.com/package/@zetachain/toolkit/v/10.0.0

done

**Review by fadeev — APPROVED** [review]

**Comment by fadeev** [comment]
@andresaiello please, review.

**Pull request merged**